### ISSUE TITLE
Feature to import data in a structured way

### ISSUE THREAD

**Issue by giomf** [erictapen/typst-invoice#6]
Hi,

im currently working on a cli that basically is a simple Customer/Invoice database and can render invoices by using typst templates.
I would like to use your template but its kinda hard to transform my data structures into typst code. 

I know that typst allows you to import data from yaml files and other templates like [invoice-maker](https://github.com/ad-si/invoice-maker) are using it. Do you think its feasible to add this feature to your template so that I could use your template to automatically render some invoices?

Thanks!

**Comment by erictapen** [comment]
So my initial version of this actually did that but over time I decided to go for the current approach. My impression was that typst isn't really made for this (yet). My problem with this workflow is, that you basically need to create two files now per invoice, a yaml/json/toml file and a .typ file.

What typst would need is some kind of command, that invokes a template from the supplied data directly, e.g. 

```console
# Results in myinvoice.pdf
typst compile --template '@preview/classy-german-invoice:0.3.1' myinvoice.toml
```

For now typst appears to me like it doesn't want to be invoked from other systems, which I consider a problem actually.

Until typst officially supports generating documents from structured data alone, I feel like providing an intermediate representation is not the way things are supposed to work. Especially when we have to maintain both representations from now on like you propose in #7.

For other systems invoking typst, I'm afraid the current situation wants you to have the data flow the other way around: Have your program generate a valid typst function call. That could be done (and tested) using some kind of templating engine or some language library that allows you to plug the primitives together and render them into a string.

**Comment by giomf** [comment]
Yes I can totally understand you it's not really nice to inject data into typst and I hope that changes but that's exactly what my CLI currently does so you don't have to do it manually yourself. 

Additionally typst is available as a [rust crate](https://docs.rs/typst/0.12.0/typst/index.html) since version 0.11.0. I haven't looked at it yet but I'm hoping that this will make it possible to call typst functions from rust code or otherwise help in the progress. But as I said, I haven't looked into it any further 

I would like to continue developing my invoice program on the basis of typst, even if typst does not currently offer the perfect API. Changes are still possible afterwards. This means that I need some way of getting data into typst, even if that means generating several documents in an additional step. As I said, the CLI does this for you.
